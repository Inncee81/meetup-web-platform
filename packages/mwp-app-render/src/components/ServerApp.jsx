import React from 'react';
import PropTypes from 'prop-types';
import StaticRouter from 'react-router-dom/StaticRouter';
import HapiCookieProvider from '@meetup/mwp-cookie/lib/HapiCookieProvider';
import PlatformApp from './shared/PlatformApp';

/**
 * A simple component to wrap the base PlatformApp with a StaticRouter
 */
class ServerApp extends React.Component {
	render() {
		const { request, h, appContext, routerContext, store, routes } = this.props;
		return (
			<StaticRouter
				basename={appContext.basename}
				location={request.raw.req.url}
				context={routerContext}
			>
				<HapiCookieProvider request={request} h={h}>
					<PlatformApp
						appContext={appContext}
						store={store}
						routes={routes}
					/>
				</HapiCookieProvider>
			</StaticRouter>
		);
	}
}

ServerApp.propTypes = {
	routes: PropTypes.array.isRequired,
	store: PropTypes.object.isRequired,
	request: PropTypes.object.isRequired,
	h: PropTypes.object.isRequired,
	routerContext: PropTypes.object.isRequired,
	appContext: PropTypes.object.isRequired,
};
ServerApp.defaultProps = {
	basename: '',
};

export default ServerApp;
