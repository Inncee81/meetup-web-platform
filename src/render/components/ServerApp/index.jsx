import React from 'react';
import PropTypes from 'prop-types';
import StaticRouter from 'react-router-dom/StaticRouter';
import PlatformApp from '../shared/PlatformApp';

/**
 * A simple component to wrap the base PlatformApp with a StaticRouter
 */
class ServerApp extends React.Component {
	render() {
		const { basename, location, staticContext, store, routes } = this.props;
		return (
			<StaticRouter
				basename={basename}
				location={location}
				context={staticContext}
			>
				<PlatformApp store={store} routes={routes} />
			</StaticRouter>
		);
	}
}

ServerApp.propTypes = {
	routes: PropTypes.array.isRequired,
	store: PropTypes.object.isRequired,
	basename: PropTypes.string.isRequired,
	location: PropTypes.string.isRequired,
	staticContext: PropTypes.object.isRequired,
};
ServerApp.defaultProps = {
	basename: '',
};

export default ServerApp;
