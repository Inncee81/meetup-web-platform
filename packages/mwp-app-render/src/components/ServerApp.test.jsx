import React from 'react';
import { shallow } from 'enzyme';
import ServerApp from './ServerApp';

describe('ServerApp', function() {
	const routes = [];
	const basename = '';
	const appWrapper = shallow(
		<ServerApp
			routes={routes}
			store={{}}
			appContext={{ basename }}
			routerContext={{}}
			location="/foo"
		/>
	);
	it('exists', function() {
		expect(appWrapper).toMatchSnapshot();
	});
});
