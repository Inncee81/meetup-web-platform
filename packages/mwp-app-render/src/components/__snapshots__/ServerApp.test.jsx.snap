// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServerApp exists 1`] = `
<StaticRouter
  basename=""
  context={Object {}}
  location="/foo"
>
  <PlatformApp
    context={
      Object {
        "basename": "",
      }
    }
    routes={Array []}
    store={Object {}}
  />
</StaticRouter>
`;
